src/backend/base/langflow/utils/connection_string_parser.py

```diff
@@ -2,10 +2,8 @@
 
 
 def transform_connection_string(connection_string):
-    db_url_name = connection_string.split("@")[-1]
-    password_url = connection_string.split(":")[-1]
-    password_string = password_url.replace(f"@{db_url_name}", "")
+    auth_part, db_url_name = connection_string.rsplit("@", 1)
+    protocol_user, password_string = auth_part.rsplit(":", 1)
     encoded_password = quote(password_string)
-    protocol_user = connection_string.split(":")[:-1]
-    transformed_connection_string = f'{":".join(protocol_user)}:{encoded_password}@{db_url_name}'
+    transformed_connection_string = "".join([":".join([protocol_user, encoded_password]), "@", db_url_name])
```

Comment:
`transformed_connection_string = f'{protocol_user}:{encoded_password}@{db_url_name}'`
would be better I think.